### PR TITLE
Bugfix: allow two detectors exactly in offline search

### PR DESCRIPTION
**Pull request by GarethCabournDavies**
The offline search was erroring when it was trying to run exclude_zerolag without any other statmap files.

This adds some logic to allow that search configuration

### CODE REVIEW

File: bin/workflows/pycbc_make_offline_search_workflow [hunk 1]
```diff
@@ -278,20 +278,29 @@ for ifocomb in ifo_sets:
...
-    final_bg_files[ordered_ifo_list] = wf.setup_exclude_zerolag(
-        workflow,
-        no_fg_exc_files[ordered_ifo_list],
-        wf.FileList(other_bg_files.values()),
-        output_dir, ordered_ifo_list,
-        tags=ctagcomb
-    )
 
-combined_bg_file = wf.setup_combine_statmap(
+    if len(ifo_sets) == 1:
```
Comment (maxtrevor) [inline bin/workflows/pycbc_make_offline_search_workflow:282]: I think this block of code all looks good

File: bin/workflows/pycbc_make_offline_search_workflow [hunk 2]
```diff
@@ -523,9 +533,10 @@ for key in final_bg_files:
         closed_page = [(snrifar_cb, ifar_cb)]
     else:
         closed_page = [(snrifar_cb,)]
-    table = wf.make_foreground_table(workflow, bg_file, hdfbank, open_dir,
-                                     singles=insps, extension='.html',
-                                     tags=bg_file.tags)
+    if len(ifo_sets) > 1:
```
Comment (maxtrevor) [inline bin/workflows/pycbc_make_offline_search_workflow:536]: what exactly is the foreground table? something in the report? are we sure its something we only want when running with 3 (or more) ifos?

File: bin/workflows/pycbc_make_offline_search_workflow [hunk 3]
```diff
@@ -278,20 +278,29 @@ for ifocomb in ifo_sets:
...
+            no_fg_exc_files[ordered_ifo_list],
+            wf.FileList(other_bg_files.values()),
+            output_dir, ordered_ifo_list,
+            tags=ctagcomb
+        )
+
+if len(ifo_sets) == 1:
+    combined_bg_file = no_fg_exc_files[ordered_ifo_list]
+else:
+    combined_bg_file = wf.setup_combine_statmap(
```
Comment (maxtrevor) [inline bin/workflows/pycbc_make_offline_search_workflow:303]: I think this block of code makes sense

File: bin/workflows/pycbc_make_offline_search_workflow [hunk 4]
```diff
@@ -523,9 +533,10 @@ for key in final_bg_files:
         closed_page = [(snrifar_cb, ifar_cb)]
     else:
         closed_page = [(snrifar_cb,)]
-    table = wf.make_foreground_table(workflow, bg_file, hdfbank, open_dir,
-                                     singles=insps, extension='.html',
-                                     tags=bg_file.tags)
+    if len(ifo_sets) > 1:
+        table = wf.make_foreground_table(workflow, bg_file, hdfbank, open_dir,
+                                         singles=insps, extension='.html',
+                                         tags=bg_file.tags)
```
Comment (maxtrevor) [inline bin/workflows/pycbc_make_offline_search_workflow:539]: what exactly is the foreground table? something in the report? are we sure its something we only want if we have 3 or more ifos?

Comment (GarethCabournDavies) [reply bin/workflows/pycbc_make_offline_search_workflow]: I've added a comment to explain this - If there is only one ifo_set, then this table has already been made, so the workflow crashes given that two jobs are doing the same thing

Comment (maxtrevor) [reply bin/workflows/pycbc_make_offline_search_workflow]: ok that makes sense

**Review by maxtrevor — APPROVED** [review]

**Comment by GarethCabournDavies** [comment]
@maxtrevor could you merge once the search test completes? Automerge doesnt take that into accoutn and it will take a while so it will be a bit late for me

**Comment by maxtrevor** [comment]
> @maxtrevor could you merge once the search test completes? Automerge doesnt take that into accoutn and it will take a while so it will be a bit late for me

Sure

**Pull request merged**